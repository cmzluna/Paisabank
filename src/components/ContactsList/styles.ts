import styled from "styled-components/native";
import BaseButton from "../BaseButton";
import { getScale } from "../../utils";
import { fontSize } from "../../theme/typography";

const { horizontalScale, verticalScale } = getScale();

const Container = styled.View`
  flex: 1;
`;

const ItemContainer = styled.TouchableOpacity`
  flex-direction: row;
  margin: 0 20px 0 20px;
  border-radius: 14px;
  background-color: #fff;
  padding: 20px;
  height: ${verticalScale(92)}px;
  margin-bottom: ${horizontalScale(26)}px;
  justify-content: space-between;
  align-items: center;
`;

const Wrapper = styled.View`
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  border: 1px solid blue;
`;

const ColumnWrapper = styled.View`
  flex: 1;
  flex-direction: column;
  margin-left: ${horizontalScale(16)}px;
`;

const InnerWrapper = styled.View`
  display: flex;
  flex: 1;
  justify-content: center;
  align-items: center;
`;

const ItemTitle = styled.Text`
  ${fontSize.xxsmall};
  text-align: center;
  font-weight: bold;
`;

const XxsmallText = styled.Text`
  ${fontSize.xxsmall};
`;
const SmallText = styled.Text`
  ${fontSize.small};
`;

const XsmallText = styled.Text`
  ${fontSize.xsmall};
  color: #aaa;
`;

interface MediumTextProps {
  color: string;
}
const MediumText = styled.Text<MediumTextProps>`
  ${fontSize.medium};
  font-weight: 500;
  color: ${(props) => props.color};
`;

const LargeText = styled.Text`
  ${fontSize.large};
`;

const XxlargeText = styled.Text`
  ${fontSize.xxlarge};
`;

const Button = styled(BaseButton)`
  background-color: #dadd33;
  border-radius: 8px;
  margin-top: 11px;
  height: 48px;
  width: 128px;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export {
  Container,
  ItemContainer,
  Wrapper,
  ItemTitle,
  InnerWrapper,
  Button,
  ColumnWrapper,
  MediumText,
  XxsmallText,
  SmallText,
  LargeText,
  XsmallText,
  XxlargeText,
};
