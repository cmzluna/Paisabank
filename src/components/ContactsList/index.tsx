/* eslint-disable @typescript-eslint/no-var-requires */
import { Container, ItemContainer, ColumnWrapper, XsmallText, LargeText } from "./styles";
import { SectionList } from "react-native";
import type { Contact } from "types";
import TextAvatar from "./components/TextAvatar";
import SectionTitle from "../SectionTitle";
import LoadingIndicator from "../LoadingIndicator";

interface ItemProps {
  item: Contact;
}

interface ContactsListProps {
  sections: Array<{
    title: string;
    data: Contact[];
  }>;
  isLoading: boolean;
}

const Item = ({ item }: ItemProps): JSX.Element => {
  const avatarInitials = item.name[0] + item.lastName[0];

  return (
    <ItemContainer>
      <TextAvatar initials={avatarInitials} />

      <ColumnWrapper testID="contact-info">
        <LargeText>
          {item.name} {item.lastName}
        </LargeText>
        <XsmallText> {item.phone}</XsmallText>
      </ColumnWrapper>
    </ItemContainer>
  );
};

const ContactsList = ({ sections, isLoading, ...props }: ContactsListProps): JSX.Element | null => {
  const renderItem = ({ item }: { item: Contact }): JSX.Element => {
    return <Item item={item} />;
  };

  if (isLoading) return <LoadingIndicator />;
  if (!sections) return null;

  return (
    <Container>
      <SectionList
        sections={sections}
        renderItem={renderItem}
        keyExtractor={(item) => item.id + item.name}
        renderSectionHeader={({ section: { title } }) => <SectionTitle title={title} showDivider />}
        {...props}
      />
    </Container>
  );
};

export default ContactsList;
