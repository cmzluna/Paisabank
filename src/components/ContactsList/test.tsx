import renderer from "react-test-renderer";
import { render } from "@testing-library/react-native";
import ContactsList from ".";

const mockedData = [
  {
    title: "Title 1",
    data: [
      {
        name: "Juan",
        lastName: "Perez",
        phone: "111-1111",
      },
    ],
  },
  {
    title: "Title 2",
    data: [
      {
        name: "Martin",
        lastName: "Rodriguez",
        phone: "222-2222",
      },
    ],
  },
  {
    title: "Title 3",
    data: [
      {
        name: "Carlos",
        lastName: "Alvarez",
        phone: "333-3333",
      },
    ],
  },
];
// const mockProps = { data: mockedData };

describe("returns error", () => {
  it("returns null when sections prop is not passed", () => {
    const ContactsListComponent = renderer.create(<ContactsList />).toJSON();
    expect(ContactsListComponent).toBeNull();
  });
});

describe("renders correctly", () => {
  it("it matches snapshot", () => {
    const ContactsListComponent = renderer
      .create(<ContactsList sections={mockedData} isLoading={false} />)
      .toJSON();
    expect(ContactsListComponent).toMatchSnapshot();
  });

  it("renders loading indicator if loading prop is true", () => {
    const { getByAccessibilityHint } = render(<ContactsList sections={mockedData} isLoading />);

    expect(getByAccessibilityHint("loading...")).toBeTruthy();
  });

  it("renders all items correctly", () => {
    const { getAllByTestId } = render(<ContactsList sections={mockedData} isLoading={false} />);

    expect(getAllByTestId("contact-info")).toHaveLength(3);
  });
});
