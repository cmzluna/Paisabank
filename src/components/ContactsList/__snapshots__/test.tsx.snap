// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly it matches snapshot 1`] = `
<View
  style={
    {
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
>
  <RCTScrollView
    data={
      [
        {
          "data": [
            {
              "lastName": "Perez",
              "name": "Juan",
              "phone": "111-1111",
            },
          ],
          "title": "Title 1",
        },
        {
          "data": [
            {
              "lastName": "Rodriguez",
              "name": "Martin",
              "phone": "222-2222",
            },
          ],
          "title": "Title 2",
        },
        {
          "data": [
            {
              "lastName": "Alvarez",
              "name": "Carlos",
              "phone": "333-3333",
            },
          ],
          "title": "Title 3",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={
      [
        0,
        3,
        6,
      ]
    }
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "marginBottom": 32,
              "marginLeft": 39,
              "marginRight": 0,
              "marginTop": 32,
            }
          }
        >
          <Text
            style={
              {
                "color": "#aaaaaa",
                "fontFamily": "Poppins-Medium",
                "fontSize": 20,
              }
            }
          >
            Title 1
          </Text>
        </View>
        <View
          style={
            {
              "borderBottomColor": "#acbac3",
              "borderBottomWidth": 1,
              "borderLeftColor": "#acbac3",
              "borderRightColor": "#acbac3",
              "borderTopColor": "#acbac3",
              "height": 1,
              "marginBottom": 35,
              "width": "100%",
            }
          }
        />
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderBottomLeftRadius": 14,
              "borderBottomRightRadius": 14,
              "borderTopLeftRadius": 14,
              "borderTopRightRadius": 14,
              "flexDirection": "row",
              "height": 136,
              "justifyContent": "space-between",
              "marginBottom": 47,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 0,
              "opacity": 1,
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#caf0ff",
                "borderBottomLeftRadius": 12,
                "borderBottomRightRadius": 12,
                "borderTopLeftRadius": 12,
                "borderTopRightRadius": 12,
                "height": 79,
                "justifyContent": "center",
                "width": 79,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#68c6e5",
                  "fontFamily": "Poppins-Medium",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "textTransform": "uppercase",
                }
              }
            >
              JP
            </Text>
          </View>
          <View
            style={
              {
                "flexBasis": 0,
                "flexDirection": "column",
                "flexGrow": 1,
                "flexShrink": 1,
                "marginLeft": 28,
              }
            }
          >
            <Text
              style={
                {
                  "fontFamily": "Poppins-Medium",
                  "fontSize": 16,
                }
              }
            >
              Juan
               
              Perez
            </Text>
            <Text
              style={
                {
                  "color": "#aaa",
                  "fontSize": 12,
                }
              }
            >
               
              111-1111
            </Text>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      />
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "marginBottom": 32,
              "marginLeft": 39,
              "marginRight": 0,
              "marginTop": 32,
            }
          }
        >
          <Text
            style={
              {
                "color": "#aaaaaa",
                "fontFamily": "Poppins-Medium",
                "fontSize": 20,
              }
            }
          >
            Title 2
          </Text>
        </View>
        <View
          style={
            {
              "borderBottomColor": "#acbac3",
              "borderBottomWidth": 1,
              "borderLeftColor": "#acbac3",
              "borderRightColor": "#acbac3",
              "borderTopColor": "#acbac3",
              "height": 1,
              "marginBottom": 35,
              "width": "100%",
            }
          }
        />
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderBottomLeftRadius": 14,
              "borderBottomRightRadius": 14,
              "borderTopLeftRadius": 14,
              "borderTopRightRadius": 14,
              "flexDirection": "row",
              "height": 136,
              "justifyContent": "space-between",
              "marginBottom": 47,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 0,
              "opacity": 1,
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#caf0ff",
                "borderBottomLeftRadius": 12,
                "borderBottomRightRadius": 12,
                "borderTopLeftRadius": 12,
                "borderTopRightRadius": 12,
                "height": 79,
                "justifyContent": "center",
                "width": 79,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#68c6e5",
                  "fontFamily": "Poppins-Medium",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "textTransform": "uppercase",
                }
              }
            >
              MR
            </Text>
          </View>
          <View
            style={
              {
                "flexBasis": 0,
                "flexDirection": "column",
                "flexGrow": 1,
                "flexShrink": 1,
                "marginLeft": 28,
              }
            }
          >
            <Text
              style={
                {
                  "fontFamily": "Poppins-Medium",
                  "fontSize": 16,
                }
              }
            >
              Martin
               
              Rodriguez
            </Text>
            <Text
              style={
                {
                  "color": "#aaa",
                  "fontSize": 12,
                }
              }
            >
               
              222-2222
            </Text>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      />
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "marginBottom": 32,
              "marginLeft": 39,
              "marginRight": 0,
              "marginTop": 32,
            }
          }
        >
          <Text
            style={
              {
                "color": "#aaaaaa",
                "fontFamily": "Poppins-Medium",
                "fontSize": 20,
              }
            }
          >
            Title 3
          </Text>
        </View>
        <View
          style={
            {
              "borderBottomColor": "#acbac3",
              "borderBottomWidth": 1,
              "borderLeftColor": "#acbac3",
              "borderRightColor": "#acbac3",
              "borderTopColor": "#acbac3",
              "height": 1,
              "marginBottom": 35,
              "width": "100%",
            }
          }
        />
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderBottomLeftRadius": 14,
              "borderBottomRightRadius": 14,
              "borderTopLeftRadius": 14,
              "borderTopRightRadius": 14,
              "flexDirection": "row",
              "height": 136,
              "justifyContent": "space-between",
              "marginBottom": 47,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 0,
              "opacity": 1,
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#caf0ff",
                "borderBottomLeftRadius": 12,
                "borderBottomRightRadius": 12,
                "borderTopLeftRadius": 12,
                "borderTopRightRadius": 12,
                "height": 79,
                "justifyContent": "center",
                "width": 79,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#68c6e5",
                  "fontFamily": "Poppins-Medium",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "textTransform": "uppercase",
                }
              }
            >
              CA
            </Text>
          </View>
          <View
            style={
              {
                "flexBasis": 0,
                "flexDirection": "column",
                "flexGrow": 1,
                "flexShrink": 1,
                "marginLeft": 28,
              }
            }
          >
            <Text
              style={
                {
                  "fontFamily": "Poppins-Medium",
                  "fontSize": 16,
                }
              }
            >
              Carlos
               
              Alvarez
            </Text>
            <Text
              style={
                {
                  "color": "#aaa",
                  "fontSize": 12,
                }
              }
            >
               
              333-3333
            </Text>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      />
    </View>
  </RCTScrollView>
</View>
`;
